### DISCUSSION TITLE
Brew Install Fails on WSL

### DISCUSSION THREAD

**Discussion by BinaryPaean** [Homebrew/discussions#4457] [Getting started] [answered]
### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

Install does the following on Windows 10 WSL + Ubuntu 20.04

```zsh
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
==> Checking for `sudo` access (which may request your password)...
[sudo] password for aaron:
==> This script will install:
/home/linuxbrew/.linuxbrew/bin/brew
/home/linuxbrew/.linuxbrew/share/doc/homebrew
/home/linuxbrew/.linuxbrew/share/man/man1/brew.1
/home/linuxbrew/.linuxbrew/share/zsh/site-functions/_brew
/home/linuxbrew/.linuxbrew/etc/bash_completion.d/brew
/home/linuxbrew/.linuxbrew/Homebrew
==> The following new directories will be created:
/home/linuxbrew/.linuxbrew/bin
/home/linuxbrew/.linuxbrew/etc
/home/linuxbrew/.linuxbrew/include
/home/linuxbrew/.linuxbrew/lib
/home/linuxbrew/.linuxbrew/sbin
/home/linuxbrew/.linuxbrew/share
/home/linuxbrew/.linuxbrew/var
/home/linuxbrew/.linuxbrew/opt
/home/linuxbrew/.linuxbrew/share/zsh
/home/linuxbrew/.linuxbrew/share/zsh/site-functions
/home/linuxbrew/.linuxbrew/var/homebrew
/home/linuxbrew/.linuxbrew/var/homebrew/linked
/home/linuxbrew/.linuxbrew/Cellar
/home/linuxbrew/.linuxbrew/Caskroom
/home/linuxbrew/.linuxbrew/Frameworks

Press RETURN/ENTER to continue or any other key to abort:
==> /usr/bin/sudo /usr/bin/install -d -o %n -g aaron -m 0755 /home/linuxbrew/.linuxbrew
/usr/bin/install: invalid user ‘%n’
Failed during: /usr/bin/sudo /usr/bin/install -d -o %n -g aaron -m 0755 /home/linuxbrew/.linuxbrew
```

I thought it was that the chomp built-in wasn't acting as expected, but fiddling with that in the install didn't seem to affect it. Any suggestions to proceed?

**Comment by ZhongRuoyu** [comment]
What's the output of all the following:

```
echo "$USER"
which id
/usr/bin/id -un
```

**Reply by BinaryPaean** [reply, marked as answer]
```zsh
~echo "$USER"
%n
~ which id
/usr/bin/id
~/usr/bin/id -un
aaron
```

Based on those results I changed the install to:
```/bin/bash -c "USER=aaron $(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"```
...and it worked. Proximal problem solved, so thanks @ZhongRuoyu - For anyone else I guess the bonus round is, why did I have to do that?

**Reply by gromgit** [reply]
> For anyone else I guess the bonus round is, why did I have to do that?

Since others have run the installer under WSL for months without issue, it's something specific to your setup.

Specifically, something in your shell startup sequence is setting `USER` to `%n`, or perhaps you set it manually and forgot about it. Either way, that's not a valid value for what `USER` is generally understood to contain.

**Reply by BinaryPaean** [reply]
I use ZSH, I doubt I even configured Bash on this machine, perhaps it is missing some init. Thanks, I'll see if there is a weird Bashrc or other obvious missing thing for future installs/scripts.
